ckan/controllers/user.py

```diff
@@ -162,10 +166,90 @@ def register(self, data=None, errors=None, error_summary=None):
 
         return self.new(data, errors, error_summary)
 
+    def _generate_internal_logs(self, log_message):
+        '''
+        This helper function is called if sending a new user registration email fails.
+        '''
+
+        logging.warning("There was an error sending the email. Writing to log file.")
+
+        log_path = os.environ['REGISTRATION_ERROR_LOG_FILE_PATH']
+
+        try:
+            with open(log_path + '/' + os.environ['REGISTRATION_ERROR_LOG_FILE_NAME'], "a") as fp:
+                try:
+                    fp.write(log_message)
+                    fp.write("\n\n")
+                except:
+                    logging.warning("There is an error writing to the log file. Dumping the output to console.")
+                    logging.warning(log_message)
+        except:
+            logging.warning("There is an error writing to the log file. Dumping the output to console.")
+            logging.warning(log_message)
+
```

Comment:
why do you have these two levels of `try: except:`? They seem to duplicate each other.

Reply:
Fixed.